[BUG] udf-example build failed: Unknown CMake command "cpm_check_if_package_already_added".

**Describe the bug**
related to https://github.com/rapidsai/rapids-cmake/pull/121 https://github.com/rapidsai/rapids-cmake/pull/111
`Unknown CMake command "cpm_check_if_package_already_added".`

```
[2021-11-15T08:55:42.594Z] [INFO]      [exec] -- Build configuration summary:
[2021-11-15T08:55:42.594Z] [INFO]      [exec] --   Generator: Unix Makefiles
[2021-11-15T08:55:42.594Z] [INFO]      [exec] --   Build type: RELEASE
[2021-11-15T08:55:42.594Z] [INFO]      [exec] --   Source directory: /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/arrow-src/cpp
[2021-11-15T08:55:42.594Z] [INFO]      [exec] --   Install prefix: /usr/local
[2021-11-15T08:55:42.594Z] [INFO]      [exec] -- Installing: /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/arrow-src/cpp/src/arrow/util/config.h
[2021-11-15T08:55:42.594Z] [INFO]      [exec] -- Installing: /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/arrow-src/cpp/src/arrow/gpu/cuda_version.h
[2021-11-15T08:55:42.594Z] [INFO]      [exec] -- CPM: cudf: adding package dlpack@0.5 (v0.5)
[2021-11-15T08:55:44.516Z] [INFO]      [exec] -- CPM: cudf: adding package libcudacxx@1.6.0 (1.6.0)
[2021-11-15T08:56:11.060Z] [INFO]      [exec] -- CPM: cudf: adding package cuco@0.0 (f0eecb203590f1f4ac4a9f1700229f4434ac64dc)
[2021-11-15T08:56:12.965Z] [INFO]      [exec] -- Configuring incomplete, errors occurred!
[2021-11-15T08:56:12.965Z] [INFO]      [exec] See also "/home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/CMakeFiles/CMakeOutput.log".
[2021-11-15T08:56:12.965Z] [INFO]      [exec] See also "/home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/CMakeFiles/CMakeError.log".
[2021-11-15T08:56:12.965Z] [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/rapids-cmake-src/rapids-cmake/cpm/find.cmake:156 (cpm_check_if_package_already_added):
[2021-11-15T08:56:12.965Z] [INFO]      [exec]   Unknown CMake command "cpm_check_if_package_already_added".
[2021-11-15T08:56:12.965Z] [INFO]      [exec] Call Stack (most recent call first):
[2021-11-15T08:56:12.965Z] [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/rapids-cmake-src/rapids-cmake/cpm/libcudacxx.cmake:73 (rapids_cpm_find)
[2021-11-15T08:56:12.965Z] [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/cuco-src/cmake/thirdparty/get_libcudacxx.cmake:19 (rapids_cpm_libcudacxx)
[2021-11-15T08:56:12.965Z] [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/cuco-src/cmake/thirdparty/get_libcudacxx.cmake:24 (find_and_configure_libcudacxx)
[2021-11-15T08:56:12.966Z] [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-rapids_build-it-UDF-native-48/udf-examples/target/cpp-build/_deps/cuco-src/CMakeLists.txt:63 (include)
[2021-11-15T08:56:12.966Z] [INFO]      [exec] 
```

--- a/udf-examples/Dockerfile
+++ b/udf-examples/Dockerfile
@@ -40,7 +40,7 @@ CUDA_VERSION_MINOR=$(echo $CUDA_VERSION | tr -d '.' | cut -c 3); \
 && add-apt-repository ppa:deadsnakes/ppa \
 && apt update -y \
 && apt install -y \
-   build-essential git wget \
+   build-essential git rsync wget \
    gcc-${GCC_VERSION} g++-${GCC_VERSION} \
    openjdk-8-jdk maven tzdata \
    # CMake dependencies

--- a/udf-examples/src/main/cpp/CMakeLists.txt
+++ b/udf-examples/src/main/cpp/CMakeLists.txt
@@ -20,7 +20,11 @@ file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-21.
      ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
 include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
 
+include(rapids-cmake)
+include(rapids-cpm)
 include(rapids-cuda)
+include(rapids-export)
+include(rapids-find)
 
 # Use GPU_ARCHS if it is defined
 if(DEFINED GPU_ARCHS)
@@ -32,19 +36,6 @@ project(UDFEXAMPLESJNI VERSION 21.12.0 LANGUAGES C CXX CUDA)
 
 option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
 option(BUILD_UDF_BENCHMARKS "Build the benchmarks" OFF)
-
-#################################################################################################
-# - CPM -----------------------------------------------------------------------------------------
-
-set(CPM_DOWNLOAD_VERSION 0.27.2)
-set(CPM_DOWNLOAD_LOCATION "${CMAKE_BINARY_DIR}/cmake/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
-
-if(NOT (EXISTS ${CPM_DOWNLOAD_LOCATION}))
-    message(STATUS "Downloading CPM.cmake")
-    file(DOWNLOAD https://github.com/TheLartians/CPM.cmake/releases/download/v${CPM_DOWNLOAD_VERSION}/CPM.cmake ${CPM_DOWNLOAD_LOCATION})
-endif()
-
-include(${CPM_DOWNLOAD_LOCATION})
 
 ###################################################################################################
 # - build type ------------------------------------------------------------------------------------
@@ -92,8 +83,9 @@ set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -w --expt-extended-lambda --expt-relax
 # Ensure CUDA runtime is dynamic despite statically linking Arrow in libcudf
 set(CUDA_USE_STATIC_CUDA_RUNTIME OFF)
 
-CPMAddPackage(NAME  cudf
-        VERSION         "21.12.00"
+rapids_cpm_init()
+rapids_cpm_find(cudf 21.12.00
+        CPM_ARGS
         GIT_REPOSITORY  https://github.com/rapidsai/cudf.git
         GIT_TAG         branch-21.12
         GIT_SHALLOW     TRUE

--- a/udf-examples/src/main/cpp/src/cosine_similarity.cu
+++ b/udf-examples/src/main/cpp/src/cosine_similarity.cu
@@ -135,10 +135,11 @@ std::unique_ptr<cudf::column> cosine_similarity(cudf::lists_column_view const& l
                     cosine_similarity_functor({lv1_data, lv2_data, lv1_offsets, lv2_offsets}));
 
   // the validity of the output is the bitwise-and of the two input validity masks
-  rmm::device_buffer null_mask = cudf::bitmask_and(cudf::table_view({lv1.parent(), lv2.parent()}));
+  auto [null_mask, null_count] = cudf::bitmask_and(cudf::table_view({lv1.parent(), lv2.parent()}));
 
   return std::make_unique<cudf::column>(cudf::data_type{cudf::type_id::FLOAT32},
                                         row_count,
                                         float_results.release(),
-                                        std::move(null_mask));
+                                        std::move(null_mask),
+                                        null_count);
 }

